### PR TITLE
added new XSIAM content items - case fields, layouts and layout rules

### PR DESCRIPTION
**Pull request by yucohen**
## Related Issues
fixes: https://jira-dc.paloaltonetworks.com/browse/CIAC-9930

## Description
Added support for new Case content items: Fields, Layouts and Layout rules.
bucket link: https://console.cloud.google.com/storage/browser/marketplace-ci-build/content/builds/CIAC-9930-xsiam_content_items/1062944/marketplacev2;tab=objects?prefix=&forceOnObjectsSortingFiltering=false

### CODE REVIEW

File: demisto_sdk/commands/common/schemas/casefield.yml [hunk 1]
```diff
@@ -0,0 +1,130 @@
...
+    type: str
+    required: true
+  name_x2:
+    type: str
+  prettyName:
+    type: str
+    required: false
+  ownerOnly:
+    type: bool
+  placeholder:
```
Comment (DeanArbel) [inline demisto_sdk/commands/common/schemas/casefield.yml:24]: Is this a real field?

File: demisto_sdk/commands/content_graph/common.py [hunk 2]
```diff
@@ -145,7 +150,7 @@ def metadata_name(self) -> str:
             return "automation"
         elif self == ContentType.INDICATOR_TYPE:
             return "reputation"
-        elif self == ContentType.LAYOUT:
+        elif self == ContentType.LAYOUT or self == ContentType.CASE_LAYOUT:
```
Comment (DeanArbel) [inline demisto_sdk/commands/content_graph/common.py:153]: You could turn this into a constant group, or just do like this
```suggestion
        elif self in (ContentType.LAYOUT, ContentType.CASE_LAYOUT):
```

File: demisto_sdk/commands/content_graph/common.py [hunk 3]
```diff
@@ -161,7 +166,7 @@ def metadata_display_name(self) -> str:
             return "Reputation"
         elif self == ContentType.MAPPER:
             return "Classifier"
-        elif self == ContentType.LAYOUT:
+        elif self == ContentType.LAYOUT or self == ContentType.CASE_LAYOUT:
```
Comment (DeanArbel) [inline demisto_sdk/commands/content_graph/common.py:169]: same here
```suggestion
        elif self in (ContentType.LAYOUT, ContentType.CASE_LAYOUT):
```

File: demisto_sdk/commands/content_graph/common.py [hunk 4]
```diff
@@ -126,7 +129,9 @@ def server_name(self) -> str:
             return "reputation"
         elif self == ContentType.INDICATOR_FIELD:
             return "incidentfield-indicatorfield"
-        elif self == ContentType.LAYOUT:
+        elif self == ContentType.CASE_FIELD:
+            return "casefield"
+        elif self == ContentType.LAYOUT or self == ContentType.CASE_LAYOUT:
```
Comment (DeanArbel) [inline demisto_sdk/commands/content_graph/common.py:134]: same here

File: demisto_sdk/commands/content_graph/parsers/case_field.py [hunk 5]
```diff
@@ -0,0 +1,51 @@
...
+
+    @property
+    def supported_marketplaces(self) -> Set[MarketplaceVersions]:
+        return {
+            MarketplaceVersions.MarketplaceV2,
+        }
+
+    def connect_to_dependencies(self) -> None:
+        """TBD in case we add case types"""
+        pass
```
Comment (DeanArbel) [inline demisto_sdk/commands/content_graph/parsers/case_field.py:51]: @yucohen can you please provide some context about this function, and why it's defined but not implemented?

File: demisto_sdk/commands/format/update_layout.py [hunk 6]
```diff
@@ -189,7 +189,11 @@ def set_layout_key(self):
             }
 
     def set_group_field(self):
-        if self.data["group"] != "incident" and self.data["group"] != "indicator":
+        if (
+            self.data["group"] != "incident"
+            and self.data["group"] != "indicator"
+            and self.data["group"] != "case"
```
Comment (DeanArbel) [inline demisto_sdk/commands/format/update_layout.py:195]: This pairing tends to repeat itself (incident + indicator + case) - usually when there's such a strong pairing it's better to unify them under a common constant, and use the constant for conditions.
Let's discuss this offline

File: .changelog/4240.yml [hunk 7]
```diff
@@ -0,0 +1,4 @@
+changes:
+- description: Added support for new XSIAM content items.
```
Comment (yucohen) [inline .changelog/4240.yml:2]:
```suggestion
- description: Added support for new XSIAM content items - Case Fields, Case layouts and Case Layout Rules
```

File: demisto_sdk/commands/common/schemas/casefield.yml (hunk 1 above)
Comment (yucohen) [reply demisto_sdk/commands/common/schemas/casefield.yml:24]: Probably legacy, removed

**Pull request merged**